### PR TITLE
[IMP]Improved hotel_restaurant module.

### PR DESCRIPTION
**Pull request by KrishnaPrajapati-SerpentCS**

### CODE REVIEW

File: hotel_restaurant/models/hotel_restaurant.py [hunk 1]
```diff
@@ -21,27 +21,60 @@ class HotelFolio(models.Model):
...
-    ismenucard = fields.Boolean('Is Menucard')
-
-
 class HotelMenucardType(models.Model):
 
     _name = 'hotel.menucard.type'
-    _description = 'Amenities Type'
+    _description = 'Food Item Type'
+
+    name = fields.Char('Name', size=64, required=True)
```
Comment (JayVora-SerpentCS) [inline hotel_restaurant/models/hotel_restaurant.py]: is size needed?

File: hotel_restaurant/models/hotel_restaurant.py [hunk 2]
```diff
@@ -389,9 +431,9 @@ def generate_kot(self):
     order_list = fields.One2many('hotel.restaurant.order.list', 'o_list',
                                  'Order List')
     tax = fields.Float('Tax (%) ')
-    amount_subtotal = fields.Float(compute='_sub_total', method=True,
-                                   string='Subtotal')
-    amount_total = fields.Float(compute='_total', method=True,
+    amount_subtotal = fields.Float(compute='_compute_amount_subtotal',
+                                   method=True, string='Subtotal')
+    amount_total = fields.Float(compute='_compute_amount_total', method=True,
```
Comment (JayVora-SerpentCS) [inline hotel_restaurant/models/hotel_restaurant.py]: Cannot we make a common method like amount_total on sale order and invoice?

File: hotel_restaurant/models/hotel_restaurant.py [hunk 3]
```diff
@@ -703,7 +744,7 @@ def on_change_item_name(self):
     kot_order_list = fields.Many2one('hotel.restaurant.kitchen.order.tickets',
                                      'Kitchen Order Tickets')
     name = fields.Many2one('hotel.menucard', 'Item Name', required=True)
-    item_qty = fields.Char('Qty', size=64, required=True)
+    item_qty = fields.Integer('Qty', size=64, required=True)
     item_rate = fields.Float('Rate', size=64)
-    price_subtotal = fields.Float(compute='_sub_total', method=True,
-                                  string='Subtotal')
+    price_subtotal = fields.Float(compute='_compute_price_subtotal',
+                                  method=True, string='Subtotal')
```
Comment (JayVora-SerpentCS) [inline hotel_restaurant/models/hotel_restaurant.py:728]: same applies here.

**Comment by JayVora-SerpentCS** [comment]
@KrishnaPrajapati-SerpentCS Can you revert the commit of alisagar?

**Comment by KrishnaPrajapati-SerpentCS** [comment]
@JayVora-SerpentCS  Yes

**Comment by JayVora-SerpentCS** [comment]
@KrishnaPrajapati-SerpentCS Please resolve the Conflicts

**Comment by KrishnaPrajapati-SerpentCS** [comment]
@JayVora-SerpentCS  Resolved the Conflicts.

**Comment by JayVora-SerpentCS** [comment]
@KrishnaPrajapati-SerpentCS Can you please take a look at the travis errors?

**Comment by KrishnaPrajapati-SerpentCS** [comment]
Yes, I see the travis errors but It's comes from hotel_reservation module and Here I pushed only hotel_restaurant module.

**Comment by omarouhari** [comment]
what remains to be done in migration and how can we validate changes?

**Comment by omarouhari** [comment]
I can not check the log. the build failed. someone would be afraid to see why?

**Comment by JayVora-SerpentCS** [comment]
See https://travis-ci.org/OCA/vertical-hotel/builds/322065859?utm_source=github_status&utm_medium=notification

**Pull request merged**